user_tools/src/spark_rapids_pytools/rapids/rapids_tool.py

```diff
@@ -592,6 +602,8 @@ def cache_all_dependencies(dep_arr: List[dict]):
                              Utils.gen_joined_str(join_elem='; ',
                                                   items=dep_list))
             self.ctxt.add_rapids_args('javaDependencies', dep_list)
+        else:
+            self.logger.warning('Dependencies were not found for the current deployment mode')
```

Comment:
If dependencies are not resolved and if we know that the JAVA cmd will fail, I think we should terminate early from python here itself.

Currently the error msg on console throws `ClassNotFoundException` and it might not be user friendly.

Reply:
Updated with raising an error